python/pyiceberg/catalog/hive.py

```diff
@@ -0,0 +1,345 @@
+        protocol = TBinaryProtocol.TBinaryProtocol(transport)
+
+        self._client = Client(protocol)
+
+    def __enter__(self) -> Client:
+        self._transport.open()
+        return self._client
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._transport.close()
+
+
+def _construct_hive_storage_descriptor(schema: Schema, location: Optional[str]) -> StorageDescriptor:
+    ser_de_info = SerDeInfo(serializationLib="org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe")
+    return StorageDescriptor(
+        _convert_schema_to_columns(schema),
+        location,
+        "org.apache.hadoop.mapred.FileInputFormat",
+        "org.apache.hadoop.mapred.FileOutputFormat",
+        serdeInfo=ser_de_info,
+    )
+
+
+def _convert_schema_to_columns(schema: Schema) -> List[FieldSchema]:
+    return [FieldSchema(field.name, _iceberg_type_to_hive_types(field.field_type), field.doc) for field in schema.fields]
+
+
+def _iceberg_type_to_hive_types(col_type: IcebergType) -> str:
+    if hive_type := hive_types.get(type(col_type)):
+        return hive_type
+    raise NotImplementedError(f"Not yet implemented column type {col_type}")
+
+
+PROP_EXTERNAL = "EXTERNAL"
+PROP_TABLE_TYPE = "table_type"
+PROP_METADATA_LOCATION = "metadata_location"
+PROP_PREVIOUS_METADATA_LOCATION = "previous_metadata_location"
+
+
+def _construct_parameters(metadata_location: str, previous_metadata_location: Optional[str] = None) -> Dict[str, Any]:
+    properties = {PROP_EXTERNAL: "TRUE", PROP_TABLE_TYPE: "ICEBERG", PROP_METADATA_LOCATION: metadata_location}
+    if previous_metadata_location:
+        properties[previous_metadata_location] = previous_metadata_location
+
+    return properties
+
+
+def _annotate_namespace(database: HiveDatabase, properties: Properties) -> HiveDatabase:
+    params = {}
+    for key, value in properties.items():
+        if key == "comment":
+            database.description = value
+        elif key == "comment":
+            database.description = value
+        else:
+            params[key] = value
+    database.parameters = params
+    return database
+
+
+class HiveCatalog(Catalog):
+    _client: _HiveClient
+
+    @staticmethod
+    def identifier_to_database(identifier: Union[str, Identifier]) -> str:
+        tuple_identifier = Catalog.identifier_to_tuple(identifier)
+        if len(tuple_identifier) != 1:
+            raise ValueError("We expect a database name, Hive does not support hierarchical namespaces")
+
+        return tuple_identifier[0]
+
+    @staticmethod
+    def identifier_to_database_table(identifier: Union[str, Identifier]) -> Tuple[str, str]:
+        tuple_identifier = Catalog.identifier_to_tuple(identifier)
+        if len(tuple_identifier) != 2:
+            raise ValueError("We expect a database.table, Hive does not support hierarchical namespaces")
+
+        return tuple_identifier[0], tuple_identifier[1]
+
+    def __init__(self, name: str, properties: Properties, url: str):
```

Comment:
Looks like you renamed the argument in `_HiveClient`, but no the one here.

Reply:
Oops, thanks!